# 1 . Identify the data type of the following values

In [1]:
type(99.9)

float

In [2]:
type("False")

str

In [3]:
type(False)

bool

In [4]:
type('0')

str

In [5]:
type(0)

int

In [6]:
type(True)

bool

In [7]:
type('True')

str

In [8]:
type([{}])

list

In [9]:
type({'a' : []})

dict

# 2. What data type would best represent the following?

### A term or phrase typed into a search box : str

### Whether or not a user is logged in : bool

### A discount amount to apply to a user's shopping cart : float

### Wheter or not a coupon code is valid : bool

### The price of a product : float

### An email address typed into a registration form : list

### Information about applicants to Codeup's data science program: dict

## 3. For each of the following code blocks:

* Read the expression and predict the evaluated results
* Execute the expression in a Python REPL

In [16]:
'1' + 2

TypeError: can only concatenate str (not "int") to str

In [17]:
6 % 4

2

In [18]:
type(6 % 4)

int

In [19]:
type(type(6 % 4))

type

In [21]:
'3 + 4 is' + 3 + 4

TypeError: can only concatenate str (not "int") to str

In [22]:
0 < 0

False

In [23]:
'False' == False

False

In [24]:
True == 'True'

False

In [25]:
5 >= -5

True

In [26]:
True or "42"

True

In [27]:
6 % 5

1

In [28]:
5 < 4 and 1 == 1

False

In [29]:
'codeup' == 'codeup' and 'codeup' == 'Codeup'

False

In [73]:
4 >= 0 and 1 !=='1'

SyntaxError: invalid syntax (464311207.py, line 1)

In [32]:
6 % 3 == 0

True

In [34]:
5 % 2 != 0

True

In [35]:
[1] + 2

TypeError: can only concatenate list (not "int") to list

In [36]:
[1] + [2]

[1, 2]

In [75]:
[1] * [2]

TypeError: can't multiply sequence by non-int of type 'list'

In [38]:
[] + [] == []

True

In [39]:
{} + {}

TypeError: unsupported operand type(s) for +: 'dict' and 'dict'

### You have rented some movies for your kids:

* The Little Mermaid for 3 days
* Brother Bear for 5 days
* Hercules for 1 day

### If the daily fee to rent a movie is 3 dollars, how much will you have to pay?



In [76]:
the_little_mermaid = 3

In [77]:
brother_bear = 5

In [78]:
hercules = 1

In [79]:
the_little_mermaid * 3 # 9 dollars 

9

In [80]:
brother_bear * 3 # 15 dollars

15

In [81]:
hercules * 3 # 3 dollars

3

In [82]:
(the_little_mermaid + brother_bear + hercules) * 3

27

In [48]:
# You will pay 27 dollars for renting these movies

## Suppose you're working as a contractor for 3 companies: Google, Amazon and Facebook.
## They pay you the following hourly rates:
* Google: 400 dollars
* Amazon: 380 dollars
* Facebook: 350 dollars

## This week you worked: 10 hours for Facebook, 6 hours for Google, and 4 hours for Amazon.

## How much will you receive in payment for this week?


In [84]:
google = 400

In [85]:
amazon = 380

In [86]:
facebook = 350

In [87]:
facebook * 10 # 3500 dollars

3500

In [88]:
google * 6 # 2400 dollars

2400

In [89]:
amazon * 4 # 1520 dollars

1520

In [90]:
(facebook * 10) + (google * 6) + (amazon * 4)

7420

#### Your payment for the week will be 7420

## A student can be enrolled in a class only if the class is not full and the class schedule does not conflict with her current schedule.



In [96]:
is_full = False
schedule_conflict = True

In [97]:
can_enroll = not schedule_conflict and not is_full
can_enroll

False

## A product offer can be applied only if a customer buys more than 2 items, and the offer has not expired. Premium members do not need to buy a specific amount of products.


In [99]:
buy_more_than_2 = False
offer_expired = False
is_premium = True

In [100]:
product_offer = (buy_more_than_2 or is_premium) and not offer_expired
product_offer 

True

In [105]:
username = 'codeup'
password = 'notastrongpassword'

In [106]:
# password_len =
password_len = len(password) >= 5
password_len

True

In [108]:
username_len = len(username) <= 20
username_len

True

In [109]:
unique_creds = password != username
unique_creds

True

In [112]:
password_len and username_len and unique_creds

True